core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -814,25 +890,23 @@ private MasterWorkerInfo findUnregisteredWorker(long workerId) {
    * @param workerId the worker id to register
    */
   @Nullable
-  private MasterWorkerInfo registerWorkerInternal(long workerId) {
+  private MasterWorkerInfo recordWorkerRegistration(long workerId) {
     for (IndexedSet<MasterWorkerInfo> workers: Arrays.asList(mTempWorkers, mLostWorkers)) {
       MasterWorkerInfo worker = workers.getFirstByField(ID_INDEX, workerId);
       if (worker == null) {
         continue;
       }
 
-      synchronized (worker) {
-        worker.updateLastUpdatedTimeMs();
```

Comment:
This update is not really needed here. I removed this line to the end of the whole register logic, so we don't need to worry about locking in this method just for updating this TS.